internal/controllers/baremetalinstance/baremetalinstance_reconciler_function.go

```
@@ -0,0 +1,501 @@
+
+const userDataSecretKey = "userdata"
+
+// FunctionBuilder contains the data and logic needed to build a function that reconciles bare metal instances.
+type FunctionBuilder struct {
+	logger     *slog.Logger
+	connection *grpc.ClientConn
+	hubCache   controllers.HubCache
+}
+
+type function struct {
+	logger                          *slog.Logger
```

golangci-lint (2.12.2)

[error] File is not properly formatted

(gofmt)